packages/lib-core/src/runtime/PluginStore.ts

```diff
@@ -281,30 +268,24 @@ export class PluginStore implements PluginStoreInterface {
    */
   private addPlugin(manifest: PluginManifest, entryModule: PluginEntryModule) {
     const pluginName = manifest.name;
+    const buildHash = manifest.buildHash ?? uuidv4();
     const reload = this.loadedPlugins.has(pluginName) || this.failedPlugins.has(pluginName);
 
-    const processedManifest: LoadedPluginManifest = {
-      name: manifest.name,
-      version: manifest.version,
-      dependencies: manifest.dependencies ?? {},
-      customProperties: manifest.customProperties ?? {},
-      baseURL: manifest.baseURL,
-      loadScripts: manifest.loadScripts,
-      registrationMethod: manifest.registrationMethod,
-      buildHash: manifest.buildHash ?? uuidv4(),
-    };
-
-    const processedExtensions = this.processExtensions(
-      pluginName,
-      processedManifest.buildHash,
-      manifest.extensions,
-      entryModule,
+    const loadedExtensions = _.cloneDeep(manifest.extensions).map<LoadedExtension>((e, index) =>
```

Comment:
Confirmed with @vojtechszocs this clone operation only occurs once when the plugin is added to the store and should have minimal performance impact. We may need to revisit this in the future if we have an overwhelming amount of extensions or plugins themselves.